src/Action.hs

```diff
@@ -167,14 +181,27 @@ type ActionExcept = ServantErr
 -- | Creates a natural transformation from Action to the servant handler monad.
 --
 -- FIXME:
--- - The ability to change the state is missing.
--- - The state should be available after run.
-mkRunAction :: (r :~> IO) -> UserState -> (Action r :~> ExceptT ServantErr IO)
-mkRunAction persistNat = \s -> Nat (run s)
+-- - The ability to change the state is missing. FIXME: which state? login modifes UserState all right
+-- - The state should be available after run. FIXME: which state? what for?
+mkRunAction :: forall r. PersistM r
+            => (r :~> IO) -> Action r :~> ExceptT ServantErr IO
+mkRunAction persistNat = Nat run
   where
-    run s = ExceptT . fmap (view _1) . runRWSTflip persistNat s . runExceptT . unAction
+    run = ExceptT . fmap (view _1) . runRWSTflip persistNat UserLoggedOut . runExceptT . (setCurrentUser >>) . unAction
```

Comment:
Yep. That was the main purpose of this overgrown PR.  To set the user state for each newly created action based on the (fake in-TVar) cookie, instead of clearing it and then auto-logging the admin user.